src/server/tx_base.cc

```diff
@@ -62,12 +62,17 @@ void RecordJournal(const OpArgs& op_args, std::string_view cmd, facade::ArgSlice
   op_args.tx->LogJournalOnShard(op_args.shard, Payload(cmd, args), shard_cnt, true);
 }
 
-void RecordExpiry(DbIndex dbid, string_view key) {
+void RecordExpiry(DbIndex dbid, string_view key, bool preempts) {
   auto journal = EngineShard::tlocal()->journal();
   CHECK(journal);
-  util::FiberAtomicGuard guard;
+  if (preempts) {
```

Comment:
if preempts == true we have a FiberAtomicGuard?

Reply:
no no you are right I fixed it, the condition was wrong